tests/s3/versioning/test_delete_object_tagging.py

```diff
@@ -348,3 +350,183 @@ def test_del_obj_tag_40436(self):
         assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
                                                     f"Expected: {put_tag} \n Actual: {get_tag}")
         LOGGER.info("ENDED: Test DELETE object tagging in a versioning enabled bucket.")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40438")
+    @CTFailOn(error_handler)
+    def test_del_obj_tag_40438(self):
+        """
+        Test DELETE object tagging for a deleted versioned object
+        """
+        LOGGER.info("STARTED: Test DELETE object tagging for a deleted versioned object")
+        ver_tag = {}
+        ver_tag.update({self.object_name: {}})
+        versions = {}
+
+        LOGGER.info("Step 1: Perform PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+        resp = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 1: Performed PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+
+        LOGGER.info("Step 2: Upload object %s before enabling versioning on bucket %s",
+                    self.object_name, self.bucket_name)
+        s3_ver_tlib.upload_version(self.s3_test_obj,
+                                   bucket_name=self.bucket_name,
+                                   file_path=self.file_path1,
+                                   object_name=self.object_name,
+                                   versions_dict=versions,
+                                   is_unversioned=True)
```

Comment:
this won't be true I guess. As versioning is enabled. Correct me @apoorva-rao if differ

Reply:
Updated accordongly.